src/metabase/api/user.clj

```diff
@@ -177,27 +188,44 @@
   Takes `query` for filtering on first name, last name, email.
   Also takes `group_id`, which filters on group id."
   [status query group_id include_deactivated]
-  {status                 (s/maybe s/Str)
-   query                  (s/maybe s/Str)
-   group_id               (s/maybe su/IntGreaterThanZero)
-   include_deactivated    (s/maybe su/BooleanString)}
-  (when (or status include_deactivated)
-    (validation/check-group-manager))
+  {status              (s/maybe s/Str)
+   query               (s/maybe s/Str)
+   group_id            (s/maybe su/IntGreaterThanZero)
+   include_deactivated (s/maybe su/BooleanString)}
+  (or
+   api/*is-superuser?*
+   (if group_id
+     (validation/check-manager-of-group group_id)
+     (validation/check-group-manager)))
```

Comment:
Both of these validation functions have a default `require-admin` as true. This flag is only relevant in the OSS scenario (so the flag should be thought of as `require-admin-in-oss`). If the user is an admin, we never have to check the group manager stuff, admins can see it. 

Otherwise, we DO have to check if the group manager is a manager of the given group_id OR if they're a manager at all when no ID is given.

Here, we rely on the `require-admin` flag to ensure that OSS doesn't slip through this. Not sure it could ever happen where a user was a group manager in an Enterprise edition, but the whole instance is downgraded to OSS, so the feature is gone but the group_manager status is still there... in any case, this will prevent that user from getting user data.